### PR TITLE
[otel] Add bazel dependency

### PR DESCRIPTION
**Pull request by yashykt**
Add bazel dependency on opentelemetry-cpp.

### CODE REVIEW

File: test/cpp/ext/filters/otel/BUILD [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    srcs = [
+        "otel_plugin_test.cc",
+    ],
+    external_deps = [
+        "gtest",
+        "otel/api",
+        "otel/sdk/src/metrics",
+    ],
+    language = "C++",
+    linkstatic = True,
```
Comment (veblush) [inline test/cpp/ext/filters/otel/BUILD:35]: Why is this required?

**Review by veblush — APPROVED** [review]

File: test/cpp/ext/filters/otel/BUILD (hunk 1 above)
Comment (yashykt) [reply test/cpp/ext/filters/otel/BUILD:35]: I dunno, I just copied it over. I'll remove it
